wiring/inc/spark_wiring_cloud.h

```diff
@@ -317,7 +318,15 @@ class CloudClass {
 
     static bool connected(void) { return spark_cloud_flag_connected(); }
     static bool disconnected(void) { return !connected(); }
-    static void connect(void) { spark_cloud_flag_connect(); }
+    static void connect(void) {
+        spark_cloud_flag_connect();
+        if (system_thread_get_state(nullptr)==spark::feature::DISABLED &&
+            SystemClass::mode() == SEMI_AUTOMATIC)
+        {
+            // Particle.connect() should be blocking in SEMI_AUTOMATIC mode when threading is disabled
+            waitUntil(connected);
```

Comment:
I wonder if we should give this a timeout value of something like 60 seconds for Wi-Fi devices and 300 seconds for Cellular devices?  Otherwise bad credentials or network issue may block user code in single threaded mode indefinitely, which no chance of user code intervention.